dataflow/spanner-io/src/main/java/com/example/dataflow/EstimateSize.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dataflow;
+
+import com.google.cloud.spanner.Struct;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * Estimates the size of the {@code Struct}.
+ */
+public class EstimateSize extends PTransform<PCollection<Struct>, PCollection<Long>> {
+
+  public static EstimateSize create() {
+    return new EstimateSize();
+  }
+
+  private EstimateSize() {
+  }
+
+  @Override
+  public PCollection<Long> expand(PCollection<Struct> input) {
+    return input.apply(ParDo.of(new EstimateStructSizeFn()));
+  }
+
+  /**
+   * Estimates the size of a Spanner row. For simplicity, arrays and structs aren't supported.
```

Comment:
why do you do this?

Reply:
What do you mean? There are two examples that estimate the logical size of the database.

Reply:
I'm asking if the comment should explain why this is being done.

Reply:
(ie. why do I need to do this?)